craft_parts/features.py

```diff
@@ -27,9 +27,16 @@
 
 @dataclasses.dataclass(frozen=True)
 class Features(metaclass=Singleton):
-    """Configurable craft-parts features."""
+    """Configurable craft-parts features.
+
+    The overlay and partitions features are mutually exclusive.
```

Comment:
What happens if we try to use them at the same time?

Reply:
Theoretically, they should be compatible so I've removed this mutually exclusive requirement.

However there is no use case for enabling both features, so I will not exhaustively test craft-parts with both features enabled.